concepts.tex

```diff
@@ -50,24 +57,74 @@
 when violated, make the function non-regular. \enternote A function that returns
 \tcode{void} is necessarily a regular function. \exitnote}
 
+{\color{newclr}
 \pnum
-\newtxt{Expressions appearing in a \techterm{requires-expression} in this document
+Expressions appearing in a \techterm{requires-expression} in this document
 are required to be equality preserving, except for those annotated with the comment
 ``not equality preserving.'' An expression so annotated may be equality preserving,
-but it is not required to be so.}
+but is not required to be so.
+
+\pnum
+An expression that may alter the value of one or more of its inputs in a manner
+observable to equality preserving expressions is said to modify those inputs.
+This document uses a notational convention to specify which expressions modify what
```

Comment:
Do you really mean for this convention to apply to the whole document, or just expressions in requires clauses?